ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/ScanResultsOverviewTable.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from 'react';
+import {
+    TableComposable,
+    Tbody,
+    Td,
+    Th,
+    Thead,
+    Tr,
+    ActionsColumn,
+    IAction,
+} from '@patternfly/react-table';
+import { format } from 'date-fns';
+
+import usePagination from 'hooks/patternfly/usePagination';
+import useTableSort from 'hooks/patternfly/useTableSort';
+
+import { ComplianceScanResultsOverview } from './types';
+import ScanResultsToolbar, { SearchFilter } from './ScanResultsToolbar';
+
+import scanResultsOverviewMockData from './MockData/complianceScanResultsOverview.json';
+
+const defaultSearchFilter = {
+    scanName: '',
+    clusterName: '',
+    profileName: '',
+};
+
+const sortFields = ['Scan Name', 'Failing Controls', 'Last Scanned'];
+const defaultSortOption = { field: 'Last Scanned', direction: 'asc' } as const;
+
+function ScanResultsOverviewTable() {
+    const { page, perPage, onSetPage, onPerPageSelect } = usePagination();
+    const { sortOption, getSortParams } = useTableSort({ sortFields, defaultSortOption });
+    const [searchFilter, setSearchFilter] = useState<SearchFilter>(defaultSearchFilter);
+    const [scanResultsOverviewData] = useState<ComplianceScanResultsOverview[]>(
+        scanResultsOverviewMockData.scanOverviews
+    );
+
+    useEffect(() => {
+        // TODO: fetch GetComplianceScanResultsOverview once implemented and update scanResultsOverviewData
+    }, [searchFilter, page, perPage, sortOption]);
```

Comment:
If any of these hook dependencies are arrays or objects, do we need `useDeepCompareEffect` hook?

For example, https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Risk/RiskTablePanel.js#L69-L92

Having said that, an alternative worth weighing, is there a reliable string representation that aggregates the dependencies? Imagine a query string **if** page address contained all of this state.

By the way, `perPage` has been in component state instead of page address as far as I know, but what do you think about search filter and sort options? Dave is a good resource which combination of hooks for that.

Reply:
We shouldn't need to perform a deep compare if we're updating the objects or arrays correctly, ensuring we create a new reference. That's why all my updates use a syntax like `setSearchFilter((prev) => ({ ...prev, profileName: profileFilter }))`. I added a `console.log` inside `useEffect` to check if everything was updating as expected and everything appeared to be working as intended unless there's something I overlooked.

Originally I was setting up the filters and pagination with the intention of including them in the URL. However I decided against it because it seemed out of place given the context. If this table were on a standalone page, integrating filters into the URL would make more sense to me. But considering the multiple other widgets that will populate the dashboard, I decided against it. That being said, I'm open to revisiting this decision but I'd like to get feedback from other folks first.

Reply:
If you decide to store these in the URL, `useURLSearch`, `useURLSort`, and `useURLPagination` are the standard ones that are most frequently used.

A bonus regarding deep compare - the values returned by those hooks are stable and only change if the underlying data does. You can safely use the objects returned as an effect dependency and the object itself will always be changed when the data changes.